server/server_test.go

```diff
@@ -0,0 +1,301 @@
+                </packages>
+            </manifest>
+        </updatecheck>
+    </app>
+</response>`
+	testCall(t, server, http.MethodPost, "", requestBody, http.StatusOK, expectedResponse)
+
+	// Other component of 2 out of date
+	requestBody = lightAndDarkThemeRequest("70.0.0", "0.0.0")
+	expectedResponse = `<response protocol="3.1" server="prod">
+    <app appid="bfdgpgibhagkpdlnjonhkabjoijopoge">
+        <updatecheck status="ok">
+            <urls>
+                <url codebase="https://s3.amazonaws.com/brave-extensions/release/bfdgpgibhagkpdlnjonhkabjoijopoge/extension_1_0_0.crx"></url>
+            </urls>
+            <manifest version="1.0.0">
+                <packages>
+                    <package name="extension_1_0_0.crx" hash_sha256="ae517d6273a4fc126961cb026e02946db4f9dbb58e3d9bc29f5e1270e3ce9834" required="true"></package>
+                </packages>
+            </manifest>
+        </updatecheck>
+    </app>
+</response>`
+	testCall(t, server, http.MethodPost, "", requestBody, http.StatusOK, expectedResponse)
+
+	// Both components need updates
+	requestBody = lightAndDarkThemeRequest("0.0.0", "0.0.0")
+	expectedResponse = `<response protocol="3.1" server="prod">
+    <app appid="ldimlcelhnjgpjjemdjokpgeeikdinbm">
+        <updatecheck status="ok">
+            <urls>
+                <url codebase="https://s3.amazonaws.com/brave-extensions/release/ldimlcelhnjgpjjemdjokpgeeikdinbm/extension_1_0_0.crx"></url>
+            </urls>
+            <manifest version="1.0.0">
+                <packages>
+                    <package name="extension_1_0_0.crx" hash_sha256="1c714fadd4208c63f74b707e4c12b81b3ad0153c37de1348fa810dd47cfc5618" required="true"></package>
+                </packages>
+            </manifest>
+        </updatecheck>
+    </app>
+    <app appid="bfdgpgibhagkpdlnjonhkabjoijopoge">
+        <updatecheck status="ok">
+            <urls>
+                <url codebase="https://s3.amazonaws.com/brave-extensions/release/bfdgpgibhagkpdlnjonhkabjoijopoge/extension_1_0_0.crx"></url>
+            </urls>
+            <manifest version="1.0.0">
+                <packages>
+                    <package name="extension_1_0_0.crx" hash_sha256="ae517d6273a4fc126961cb026e02946db4f9dbb58e3d9bc29f5e1270e3ce9834" required="true"></package>
+                </packages>
+            </manifest>
+        </updatecheck>
+    </app>
+</response>`
+	testCall(t, server, http.MethodPost, "", requestBody, http.StatusOK, expectedResponse)
+
+	// Unkonwn extension ID goes to Google server
+	requestBody = extensiontest.ExtensionRequestFnFor("aaaaaaaaaaaaaaaaaaaa")("0.0.0")
+	expectedResponse = ""
+	testCall(t, server, http.MethodPost, "", requestBody, http.StatusTemporaryRedirect, expectedResponse)
+
+	// Make sure a huge request body does not crash the server
+	data := make([]byte, 1024 * 1024 * 11) // 11 MiB
+	rand.Read(data)
```

Comment:
@bbondy CI is failing due to a few lint issues. For one you will need to run `gofmt -s` on this file, for the other you can add `// #nosec` on the end of this line to disable the gas (gosec) linter